compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -690,6 +696,12 @@ fn resolve_trait_impls(
             errors,
         );
 
+        let mut new_resolver =
+            Resolver::new(interner, &path_resolver, &context.def_maps, trait_impl.file_id);
+        new_resolver.set_self_type(Some(self_type.clone()));
+
+        check_methods_signatures(&mut new_resolver, &impl_methods, trait_id, errors);
```

Comment:
We should type check during the type checking pass, not the name resolution pass. This usage is probably fine since it is at the end after resolving but we should still move it to type checking for separation of concerns.

Reply:
I understand, but can we leave this for after the two trait bounds PRs are merged? Right now we rely on `resolver.get_self_type()` to get the self_type, which we don't store anywhere else. The next PR adds a HIR struct for a TraitImpl, and it will be much easier to move this out if we have access to that.